uast/transformer/ops.go

```diff
@@ -746,11 +746,23 @@ func (op opScope) Construct(st *State, n nodes.Node) (nodes.Node, error) {
 // Field is an operation on a specific field of an object.
 type Field struct {
 	Name string // name of the field
-	// Optional can be set to make a field optional. Provided string is used as a variable name to the state of the field.
-	// Note that "optional" means that the field may not exists in the object, and it does not mean that the field can be nil.
+	// Optional can be set to make a field optional. Provided string is used as a variable
+	// name to the state of the field. Note that "optional" means that the field may not
+	// exists in the object, and it does not mean that the field can be nil.
 	// To handle nil fields, see Opt operation.
 	Optional string
-	Op       Op // operation used to check/construct the field value
+	// Drop the field if it exists. If Op is set, it will be called to check the value
+	// before dropping it. If the check fails, the whole transform will be canceled.
+	// This is the preferred way for dropping fields. Avoid using Drop without Op.
```

Comment:
Would it make sense to consider this an error in the validation spec? If we make it always an error to have drop=true with op=nil, that would make tests fail consistently if we forget to set them.

Not crucial, IMO, but I worry a little about how this would manifest in tests.

Reply:
No, it's not an error, it will just assume `Op: Any()` in this case.

See https://github.com/bblfsh/sdk/pull/369#discussion_r259027330

Reply:
Right, I guess what I'm suggesting is maybe we should require the `Any()` to be added explicitly if that's what you want—then there's no chance we get it "wrong".

Reply:
Yeah, this sounds like a better alternative, although a bit more verbose. But it's fine for the "wrong" case of using `Any` to drop things.

Reply:
Yeah, since we suggest not using Drop unless necessary, I think the extra friction will actually be useful in this situation :)